src/app/time-select/time-select.component.ts

```diff
@@ -0,0 +1,189 @@
+import {SingleCalendarValue} from '../common/types/single-calendar-value';
+import {ECalendarType} from '../common/types/calendar-type-enum';
+import {
+  Component,
+  OnInit,
+  Input,
+  Output,
+  EventEmitter,
+  OnChanges,
+  SimpleChanges,
+  forwardRef,
+  HostBinding
+} from '@angular/core';
+import {TimeSelectService, TimeUnit} from './time-select.service';
+import * as moment from 'moment';
+import {Moment} from 'moment';
+import {ITimeSelectConfig} from './time-select-config.model';
+import {
+  NG_VALUE_ACCESSOR,
+  NG_VALIDATORS,
+  ControlValueAccessor,
+  Validator,
+  ValidationErrors,
+  FormControl
+} from '@angular/forms';
+import {CalendarValue} from '../common/types/calendar-value';
+import {UtilsService} from '../common/services/utils/utils.service';
+import {IMonthCalendarConfig} from '../month-calendar/month-calendar-config';
+import {IMonth} from '../month-calendar/month.model';
+
+@Component({
+  selector: 'dp-time-select',
+  templateUrl: 'time-select.component.html',
+  styleUrls: ['time-select.component.less'],
+  providers: [
+    TimeSelectService,
+    {
+      provide: NG_VALUE_ACCESSOR,
+      useExisting: forwardRef(() => TimeSelectComponent),
+      multi: true
+    },
+    {
+      provide: NG_VALIDATORS,
+      useExisting: forwardRef(() => TimeSelectComponent),
+      multi: true
+    }
+  ]
+})
+export class TimeSelectComponent implements OnInit, OnChanges, ControlValueAccessor, Validator {
+
+  @Input() config: ITimeSelectConfig;
+  @Input() displayDate: SingleCalendarValue;
+  @Input() minDate: Moment;
+  @Input() maxDate: Moment;
+  @Input() minTime: Moment;
+  @Input() maxTime: Moment;
+  @HostBinding('class') @Input() theme: string;
+
+  CalendarType = ECalendarType;
+  isInited: boolean = false;
+  componentConfig: ITimeSelectConfig;
+  _selected: Moment;
+  inputValue: CalendarValue;
+  inputValueType: ECalendarValue;
+  validateFn: (inputVal: CalendarValue) => {[key: string]: any};
+
+  hours: string;
+  minutes: string;
+  seconds: string;
+  meridiem: string;
+
+  set selected(selected: Moment) {
+    this._selected = selected;
+    this.calculateTimeParts(selected);
+    this.onChangeCallback(this.processOnChangeCallback(selected));
+  }
+
+  get selected(): Moment {
+    return this._selected;
+  }
+
+  constructor(public timeSelectService: TimeSelectService,
+              public utilsService: UtilsService) {
+  }
+
+  ngOnInit() {
+    this.isInited = true;
+    this.init();
+    this.initValidators();
+  }
+
+  init() {
+    this.componentConfig = this.timeSelectService.getConfig(this.config);
+    this.selected = this.selected || moment();
+    this.inputValueType = this.utilsService.getInputType(this.inputValue, false);
+  }
+
+  ngOnChanges(changes: SimpleChanges) {
+    if (this.isInited) {
+      const {minDate, maxDate, minTime, maxTime} = changes;
+      this.init();
+
+      if (minDate || maxDate || minTime || maxTime) {
```

Comment:
That check doesn't make sense in the `initValidators` method.  It's checking to see if those properties have been changed - which only makes sense in `ngOnChanges`. See https://github.com/vlio20/angular-datepicker/blob/master/src/app/day-calendar/day-calendar.component.ts#L119